java-checks/src/main/java/org/sonar/java/checks/ReplaceLambdaByMethodRefCheck.java

```diff
@@ -106,6 +127,23 @@ private void visitLambdaExpression(LambdaExpressionTree tree) {
     return Optional.empty();
   }
 
+  private static Optional<String> getTypeCast(@Nullable Tree statement, LambdaExpressionTree tree) {
+    if (statement == null) {
+      return Optional.empty();
+    }
+    Tree expr = statement;
+    if (expr.is(Tree.Kind.PARENTHESIZED_EXPRESSION)) {
```

Comment:
I feel this check defeats a bit the purpose of using `ExpressionUtils.skipParentheses(...);`, Won't it be possible to already arrive here which a (Nullable) ExpressionTree instead? It would make things easier.